### PR TITLE
Fit continuum over multiple windows.

### PR DESCRIPTION
**Pull request by ibusko**
This PR is in lieu of PR #681 , which got busted when my fork got busted. Discussions associated to this work can be found in #681 .

The implementation was actually simpler than it seemed at first. The code to fit a single model (the continuum) over multiple regions is already in place. To use it, it is just a matter of passing a single window to `fit_continuum`, using a single instance of `SpectralRegion`, but initialized with a tuple of regions. It is not enough to pass a sequence of tuples to the `window` parameter. An explicit instance of `SpectralRegion` does the trick.

### CODE REVIEW

**Comment by nmearl** [comment]
Just to clarify, is the behavior different when doing

```python
fit_continuum(spectrum, window=[(0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)])
```

Vs.

```python
fit_continuum(spectrum, window=SpectralRegion((0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)))
```

**Comment by ibusko** [comment]
Yes, it’s different. In the first case, it only picks up the first tuple.


On Jul 2, 2020, at 1:13 PM, Nicholas Earl <notifications@github.com<mailto:notifications@github.com>> wrote:



Just to clarify, is the behavior different when doing

fit_continuum(spectrum, window=[(0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)])

Vs.

fit_continuum(spectrum, window=SpectralRegion((0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)))

**Comment by nmearl** [comment]
I think we want to have the first case be a shorter version of the second. That is, if provided `window=[(0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)]`, we should internally convert that to `window=SpectralRegion((0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um))`.

**Comment by ibusko** [comment]
Sounds OK by me. I will work on it.

On Jul 2, 2020, at 1:19 PM, Nicholas Earl <notifications@github.com<mailto:notifications@github.com>> wrote:



I think we want to have the first case be a shorter version of the second. That is, if provided window=[(0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)], we should internally convert that to window=SpectralRegion((0.*u.um, 5.*u.um),(8.*u.um, 10.*u.um)).

**Comment by PatrickOgle** [comment]
I tested this using two different formats for the fit windows:
fitwindows1=[(5125.*u.AA, 5175.*u.AA),(5225.*u.AA, 5275.*u.AA)]
and
region1 = SpectralRegion(5125.*u.AA, 5175.*u.AA)
region2 = SpectralRegion(5225.*u.AA, 5275.*u.AA)
fitwindows3=[region1,region2]

This plot shows the results for one particular realization of the noise:
fit_continuum_2inputs_v1

The fit looks pretty good, and is the same regardless of the format of the windows (SpectralRegion or list of tuples).

However, I note that the fit is not always good, depending on the noise, which makes me suspicious that the fitting algorithm for fit_continuum is not robust.  This deserves a separate issue, and should not block this PR. Here is an example of a poor continuum fit, using the same windows but different random noise:
fit_continuum_2inputs_v2

File: docs/fitting.rst [hunk 1]
```diff
@@ -678,6 +678,36 @@ fitted continuum, which returns a new object:
...
+    y += np.random.normal(0., 0.2, x.shape)
+    y += 3.2 * np.exp(-0.5 * (x - 5.6)**2 / 4.8**2)
+
+    spectrum = Spectrum1D(flux=y * u.Jy, spectral_axis=x * u.um)
+
+    region = [(4.*u.um, 5.*u.um), (8.*u.um, 10.*u.um)]
+
+    fitted_continuum = fit_continuum(spectrum, window=region)
+
+
```
Comment (dtn5ah) [inline docs/fitting.rst:710]:
```suggestion
y_fit = fitted_continuum(x*u.um)

plt.plot(x,y)
plt.plot(x,y_fit)
plt.title('Continuum Fitting')
plt.grid(True)
```

**Comment by nmearl** [comment]
@PatrickOgle Can you open an issue describing your concerns with the robustness of the `fit_continuum` function? I'm going to go ahead and merge this.

**Pull request merged**